airgun/widgets.py

```diff
@@ -1085,12 +1085,24 @@ class ConfirmationDialog(Widget):
     discard_dialog = Text(
         ".//button[contains(@ng-click, 'cancel') and @class='close']")
 
+    def _check_is_displayed(self, value=None):
+        """ This is to check if dialog is displayed """
+        action = self.cancel_dialog.is_displayed if value else self.confirm_dialog.is_displayed
```

Comment:
I believe you can get away with checking visibility of same element in all cases.
If you want this parametrized, I recommend passing in reference to element:
```python
def _check_is_displayed(self, elem):
    wait_for(
        lambda: elem.is_displayed,
        timeout=30,
        delay=1,
        logger=self.logger
    )

def confirm(self):
    self._check_is_displayed(self.confirm_dialog)
```

Reply:
I think it's good in both way .. let me replace with yours one.

Reply:
it's there now ..